fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/fluid/FluidPreconditions.java

```
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.fluid;
+
+import com.google.common.base.Preconditions;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.Fluids;
+
+/**
+ * Preconditions for fluid transfer.
+ */
+public class FluidPreconditions {
+	public static void notEmpty(Fluid fluid) {
+		if (fluid == Fluids.EMPTY) {
+			throw new IllegalArgumentException("Fluid may not be empty.");
+		}
+	}
+
+	public static void notEmptyNotNegative(Fluid fluid, long amount) {
+		FluidPreconditions.notEmpty(fluid);
+		Preconditions.checkArgument(amount >= 0);
+	}
```

Why use Preconditions for non-negative, but not for non-empty?

---

Fair point, I should probably throw the exception directly so that it contains the amount.